animation_nodes/math/rotation_conversion.pyx

```diff
@@ -122,3 +127,119 @@ cdef void normalizedAxisCosAngleToMatrix(Matrix3_or_Matrix4* m, Vector3* axis, f
     if Matrix3_or_Matrix4 is Matrix4:
         m.a14, m.a24, m.a34 = 0, 0, 0
         m.a41, m.a42, m.a43, m.a44 = 0, 0, 0, 1
+
+# Euler to Quaternion
+##########################################################    
+        
+# https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+cdef euler3ToQuaternion(Quaternion* q, Euler3 *e):
+    cdef float cx = cos(e.x * 0.5)
+    cdef float sx = sin(e.x * 0.5)
+    cdef float cy = cos(e.y * 0.5)
+    cdef float sy = sin(e.y * 0.5)
+    cdef float cz = cos(e.z * 0.5)
+    cdef float sz = sin(e.z * 0.5)
+    
+    cdef float cxcy = cx * cy
+    cdef float sxcy = sx * cy
+    cdef float sxsy = sx * sy
+    cdef float cxsy = cx * sy
+    
+    q.w = cxcy * cz + sxsy * sz
+    q.x = sxcy * cz - cxsy * sz
+    q.y = cxsy * cz + sxcy * sz
+    q.z = cxcy * sz - sxsy * cz
+    
+# Quaternion to Euler
+##########################################################
+
+# https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+cdef quaternionToEuler3(Euler3 *e, Quaternion *q):
+    quaternionNormalize_InPlace(q)
```

Comment:
Lets assume all quaternions are normalized. So just remove all the necessary `quaternionNormalize_InPlace`.